src/CashFlows.py

```diff
@@ -1011,10 +1015,15 @@ def extendParameters(self, toExtend, t):
     # for capex, both the Driver and Alpha are nonzero in year 1 and zero thereafter
     for name, value in toExtend.items():
       if name.lower() in ['alpha', 'driver']:
-        if utils.isAFloatOrInt(value) or (len(value) == 1 and utils.isAFloatOrInt(value[0])):
+        if "pyomo.core.expr" in str(type(value)):
```

Comment:
I'm still not sure what the best way to act on this is; I really dislike string-type checking, but I don't really want to add pyomo as a dependency to TEAL, either.

Reply:
I found some interesting options, but it occurred to me the option may be even easier: can we check the opposite? i.e. if it's a float (or array of floats), then treat it as such; else treat it like pyomo (or anything else, since it uses "object")

Reply:
That is a nearly perfect answer! If further distinction is required later, then we'll leave it to future PRs.

Reply:
I hit an unexpected error. Apparently, len() for a pyomo expressionis not valid. The evaluation 'len(value) == 1 ' is triggering the error. Would it be acceptable to remove this condition and afterwards check if the length is one?
Like:
if utils.isAFloatOrInt(value):
  if (len(value) == 1 and utils.isAFloatOrInt(value[0])):
    ........

Reply:
Would this be doubly checking too much?

Reply:
Hm, interesting. We should probably do a type check first and then only check length if it's a np.array or float type; otherwise we treat it as a pyomo object. Data checking here might get tricky; let me know and we can look at it together if useful.